packages/scene-composer/src/components/panels/scene-settings/MatterportIntegration.tsx

```diff
@@ -88,20 +102,28 @@ export const MatterportIntegration: React.FC = () => {
   const onMatterportModelIdChange = useCallback(
     (event) => {
       const value = event.detail.value;
-      if (value !== matterportModelId) {
-        setMatterportModelId(value);
+      if (value !== matterportModelId.current) {
+        matterportModelId.current = value;
         setDirty(true);
       }
     },
-    [setMatterportModelId, setDirty],
+    [setDirty],
   );
 
   useEffect(() => {
```

Comment:
i'm not convinced we need an effect to solve this. why do we need to use effect to update these?